File: src/main/java/LeetCode/Easy/MajorityElement.java
package LeetCode.Easy;

import java.util.HashMap;

public class MajorityElement {
    public static int majorityElement(int[] nums) {
        HashMap<Integer, Integer> frequencyMap = new HashMap<Integer, Integer>();
        for (int num : nums) {
            if (frequencyMap.get(num) == null) {
                frequencyMap.put(num, 1);
            } else {
                frequencyMap.put(num, frequencyMap.get(num) + 1);
            }
            if (frequencyMap.get(num)>nums.length/2) return num;
        }
        return 0;
    }
}

File: src/main/java/LeetCode/JavaScript/Easy/Majority Element.js
let freqObj = {};

var majorityElement = function(nums) {
    let freqObj = {};

    for (number in nums) {
        if (isNaN(freqObj[number])) {
            freqObj[number] = 1;
        } else if (freqObj[number]>nums.length/2) {
            return number;
        } else {
            freqObj[number] += 1;
        }
    }

    console.log(freqObj);

    return -1;
};